pandas/tests/reshape/merge/test_merge_asof.py

```diff
@@ -1224,6 +1224,49 @@ def test_merge_on_nans(self, func, side):
             else:
                 merge_asof(df, df_null, on="a")
 
+    def test_by_nullable(self, any_numeric_ea_dtype):
+        # Note: this test passes if instead of using pd.array we use
```

Comment:
Maybe note that the outcome of https://github.com/pandas-dev/pandas/issues/32306 may be relevant to the expected behavior